sdks/cpp/lite/include/NamedChoiceConstraint.h

```diff
@@ -0,0 +1,99 @@
+#pragma once
+
+/**
+ * @file NamedChoiceConstraint.h
+ * @brief A constraint that checks if a value is within a named choice
+ * @author isaac.robert@rossvideo.com
+ * @date 2024-07-25
+ * @copyright Copyright (c) 2024 Ross Video
+ */
+
+#include <common/include/IConstraint.h>
+#include <google/protobuf/message_lite.h>
+#include <lite/include/PolyglotText.h>
+
+using catena::common::IConstraint;
```

Comment:
these 2 lines are problematic. reason is that they pollute any file that includes this header to make `IConstraint` be the `catena::common::IConstraint` class.

You can get the same benefit of typename length reduction but with the added one of encapsulating the typedef by putting code like this in the actual class definition:

```cpp
public:
  using IConstraint = catena::common::IConstraint;
  using PolyglotText = catena::lite::PolyglotText;
```

this should be a separate `public:` section of the class. See `IParam.h` for an example.